src/main/webapp/app/modeling-assessment-editor/modeling-assessment-conflict/modeling-assessment-conflict.component.ts

```diff
@@ -1,45 +1,76 @@
-    styleUrls: ['./modeling-assessment-conflict.component.scss'],
+    styleUrls: ['./modeling-assessment-conflict.component.scss', '../modeling-assessment-editor.component.scss'],
 })
 export class ModelingAssessmentConflictComponent implements OnInit, AfterViewInit {
     model: UMLModel;
+    mergedFeedbacks: Feedback[];
+    currentFeedbacksCopy: Feedback[];
     modelHighlightedElementIds: Set<string>;
-    conflictIndex = 0;
+    highlightColor: string;
     user: User;
 
     currentConflict: Conflict;
     conflictingResult: ConflictingResult;
     conflictingModel: UMLModel;
     conflictingModelHighlightedElementIds: Set<string>;
+    conflicts: Conflict[];
+    conflictResolutionStates: ConflictResolutionState[];
+    conflictIndex = 0;
+    conflictsAllHandled = false;
+    modelingExercise: ModelingExercise;
+    submissionId: number;
 
-    @Input() modelingExercise: ModelingExercise;
-    @Input() conflicts: Conflict[] = [];
-
-    constructor(private route: ActivatedRoute, private modelingSubmisionService: ModelingSubmissionService, private accountService: AccountService) {}
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private route: ActivatedRoute,
+        private router: Router,
+        private modelingSubmissionService: ModelingSubmissionService,
+        private modelingAssessmentService: ModelingAssessmentService,
+        private accountService: AccountService,
+    ) {}
 
     ngOnInit() {
-        if (this.conflicts) {
-            this.updateSelectedConflict();
-            this.model = JSON.parse((this.currentConflict.result.submission as ModelingSubmission).model);
-        }
+        this.route.params.subscribe(params => {
+            this.submissionId = Number(params['submissionId']);
+            this.conflicts = this.modelingAssessmentService.getLocalConflicts(this.submissionId);
+            if (this.conflicts && this.conflicts.length > 0) {
+                this.initComponent();
+                this.model = JSON.parse((this.currentConflict.causingConflictingResult.result.submission as ModelingSubmission).model);
+                this.modelingExercise = this.currentConflict.causingConflictingResult.result.participation.exercise as ModelingExercise;
+            } else {
+                this.conflicts = undefined;
+                this.jhiAlertService.error('modelingAssessmentEditor.messages.noConflicts');
+            }
+        });
         this.accountService.identity().then(value => (this.user = value));
     }
 
     ngAfterViewInit() {
         this.setSameWidthOnModelingAssessments();
     }
 
+    initComponent() {
+        this.mergedFeedbacks = [...this.conflicts[0].causingConflictingResult.result.feedbacks];
+        this.currentFeedbacksCopy = [...this.conflicts[0].causingConflictingResult.result.feedbacks];
+        this.conflictResolutionStates = [...this.conflictResolutionStates.map(() => ConflictResolutionState.UNHANDLED)];
```

Comment:
Doesn't this throw an error as `this.conflictResolutionStates` is initially undefined?

Reply:
yep fixed it :)